pkg/controller/admissionchecks/multikueue/fswatch_test.go

```diff
@@ -0,0 +1,460 @@
+			},
+			wantEventsForClusters: set.New("c1"),
+		},
+		"single cluster missing file": {
+			clusters: map[string]string{
+				"c1": "c1.kubeconfig",
+			},
+			opFnc: func(basePath string) error {
+				return os.WriteFile(path.Join(basePath, "c1.kubeconfig"), []byte("123456"), 0666)
+			},
+			wantEventsForClusters: set.New("c1"),
+		},
+		"single existing file deleted": {
+			prepareFnc: func(basePath string) error {
+				return os.WriteFile(path.Join(basePath, "c1.kubeconfig"), []byte("123"), 0666)
+			},
+			clusters: map[string]string{
+				"c1": "c1.kubeconfig",
+			},
+			opFnc: func(basePath string) error {
+				return os.Remove(path.Join(basePath, "c1.kubeconfig"))
+			},
+			wantEventsForClusters: set.New("c1"),
+		},
+		"single cluster file rename to": {
+			prepareFnc: func(basePath string) error {
+				return os.WriteFile(path.Join(basePath, "c1.kubeconfig.old"), []byte("123"), 0666)
+			},
+			clusters: map[string]string{
+				"c1": "c1.kubeconfig",
+			},
+			opFnc: func(basePath string) error {
+				return os.Rename(path.Join(basePath, "c1.kubeconfig.old"), path.Join(basePath, "c1.kubeconfig"))
+			},
+			wantEventsForClusters: set.New("c1"),
+		},
+		"single cluster file rename from": {
+			prepareFnc: func(basePath string) error {
+				return os.WriteFile(path.Join(basePath, "c1.kubeconfig"), []byte("123"), 0666)
+			},
+			clusters: map[string]string{
+				"c1": "c1.kubeconfig",
+			},
+			opFnc: func(basePath string) error {
+				return os.Rename(path.Join(basePath, "c1.kubeconfig"), path.Join(basePath, "c1.kubeconfig.new"))
+			},
+			wantEventsForClusters: set.New("c1"),
+		},
+		"single crete link": {
+			prepareFnc: func(basePath string) error {
+				return os.WriteFile(path.Join(basePath, "c1.kubeconfig.src"), []byte("123"), 0666)
+			},
+			clusters: map[string]string{
+				"c1": "c1.kubeconfig",
+			},
+			opFnc: func(basePath string) error {
+				return os.Symlink(path.Join(basePath, "c1.kubeconfig.src"), path.Join(basePath, "c1.kubeconfig"))
+			},
+			wantEventsForClusters: set.New("c1"),
+		},
+		"single remove link": {
+			prepareFnc: func(basePath string) error {
+				if err := os.WriteFile(path.Join(basePath, "c1.kubeconfig.src"), []byte("123"), 0666); err != nil {
+					return err
+				}
+				return os.Symlink(path.Join(basePath, "c1.kubeconfig.src"), path.Join(basePath, "c1.kubeconfig"))
+			},
+			clusters: map[string]string{
+				"c1": "c1.kubeconfig",
+			},
+			opFnc: func(basePath string) error {
+				return os.Remove(path.Join(basePath, "c1.kubeconfig"))
+			},
+			wantEventsForClusters: set.New("c1"),
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			basePath := t.TempDir()
+			if tc.prepareFnc != nil {
+				if err := tc.prepareFnc(basePath); err != nil {
+					t.Fatalf("unexpected prepare error: %s", err)
+				}
+			}
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
I'm wondering if there is a room for flaking. Could we increase the timeout to say 5s?

Reply:
We are orders of magnitude away from that, (the event is generated in less 0.5ms)

Reply:
I see we can try with 1s, but I had some experience when the CI builds were much slower than locally